app/controllers/submitted_order_selection.js

```diff
@@ -15,6 +15,32 @@ export default Ember.Controller.extend(asyncTasksMixin, {
     return this.get("cart.isEmpty");
   },
 
+  submittableOrders: Ember.computed("model.@each.state", function() {
+    let submittableOrders = this.get("model").filter(
+      this.isPackageSubmittableOrder
+    );
+    if (submittableOrders.length) {
+      return submittableOrders;
+    } else {
+      this.redirectToRequestPurpose();
```

Comment:
redirection inside a computed property is a time bomb.  You can move that to the route, `submittableOrders` should behave like a readonly variable, thus having no side effects